include/deal.II/lac/linear_operator.h

```diff
@@ -377,58 +419,65 @@ operator*(typename Range::value_type  number,
     std::is_convertible<typename Range::value_type, typename Domain::value_type>::value,
     "Range and Domain must have implicitly convertible 'value_type's");
 
-  LinearOperator<Range, Domain> return_op = op;
+  if (op.is_null_operator)
+    {
+      return op;
+    }
+  else
```

Comment:
Shouldn't we add a check here that if `number==0`, then we return a brand new null_operator, i.e.:

~~~
else if(number == 0) {
 return null_operator(op.reinit_domain_vector, op.reinit_range_vector);
} else { 
...
~~~
and remove all further checking on `number != 0`, down below?